frontend/src/metabase-lib/metadata/utils/models.unit.spec.js

```diff
@@ -13,11 +20,18 @@ import {
   getDatasetMetadataCompletenessPercentage,
 } from "metabase-lib/metadata/utils/models";
 
+const noNestedQueriesDB = createMockDatabase({
+  id: SAMPLE_DB_ID + 1,
+  features: [],
+});
+
+const metadata = createMockMetadata({
+  databases: [createSampleDatabase(), noNestedQueriesDB],
+});
+
 describe("data model utils", () => {
-  const DB_WITHOUT_NESTED_QUERIES_SUPPORT = new Database({
-    ...createMockDatabase(),
-    features: [],
-  });
+  const ordersTable = metadata.table(ORDERS_ID);
+  const DB_WITHOUT_NESTED_QUERIES_SUPPORT = new Database(noNestedQueriesDB);
```

Comment:
Types in `metadata` classes are currently incorrect, but when I fix all them similar to schemas https://github.com/metabase/metabase/pull/30694 there would be a type error with passing api type to the entity type constructor. Because the entity actually required a normalized type. The approach we suggest in the testing docs is to go through metadata creation and not create entities directly.

You already have `noNestedQueriesDB` in metadata, so you only need to get the entity instance via `checkNotNull(metadata.database(noNestedQueriesDB.id))`